pkg/compiler/validators/bindings.go

```diff
@@ -91,6 +91,21 @@ func validateBinding(w c.WorkflowBuilder, nodeID c.NodeID, nodeParam string, bin
 			errs.Collect(errors.NewMismatchingTypesErr(nodeID, nodeParam, literalType.String(), expectedType.String()))
 		}
 
+		if expectedType.GetEnumType() != nil {
+			v := binding.GetScalar().GetPrimitive().GetStringValue()
```

Comment:
do you need to do any nil checks before dereferencing the string value?

Reply:
furthermore, what's the context of this call? does a binding always have an assigned value?

Reply:
yes, this is static binding. I was thinking you should not need a nil check, but I think you are right, it is possible that someone may try to add an int to an enum

Reply:
Actually I do not need the nil check. the AreTypesCastable will fail for any non string scalar when the source type is Enum